src/compiler/core.ts

```diff
@@ -788,6 +791,249 @@ namespace ts {
         getSignatureConstructor: () => <any>Signature
     };
 
+    /**
+      * Creates an object used to navigate the ancestor's of a node by following parent pointers.
+      * @param currentNode The current node for the navigator.
+      */
+    export function createParentNavigator(currentNode: Node): ParentNavigator {
+        /** Traverses the parent pointers for the current node to find the root node. */ 

```

Comment:
If the node is synthesized and doesn't have a parent pointer. The `getRoot` and `moveToRoot` methods aren't used though, so I may remove them.